toxcore/group_chats.c

```diff
@@ -4767,10 +4767,28 @@ int gc_send_message(const GC_Chat *chat, const uint8_t *message, uint16_t length
 
     const uint8_t packet_type = type == GC_MESSAGE_TYPE_NORMAL ? GM_PLAIN_MESSAGE : GM_ACTION_MESSAGE;
 
-    if (!send_gc_broadcast_message(chat, message, length, packet_type)) {
+    uint16_t length_raw = length + GC_MESSAGE_PSEUDO_ID_SIZE;
+    uint8_t *message_raw = (uint8_t *)calloc(1, length_raw);
+    if (message_raw == nullptr) {
         return -5;
     }
 
+    uint32_t pseudo_msg_id = random_u32(chat->rng);
+    net_pack_u32(message_raw, pseudo_msg_id);
+
+    uint8_t *message_text = message_raw + (GC_MESSAGE_PSEUDO_ID_SIZE);
+    memcpy(message_text, message, length);
```

Comment:
Using a third variable is unnecessary and error prone; just do `memcpy(message_raw + GC_MESSAGE_PSEUDO_ID_SIZE, message, length);`

Reply:
done